maturity-matrix.yaml

```diff
@@ -0,0 +1,161 @@
+# backwards-incompatible changes should bump the major version number).
+apiMaturityLevels:
+  # "unimplemented" APIs do not exist yet for the component/language.
+  - unimplemented
+  # "alpha" maturity APIs can change in incompatible ways at any time.
+  - alpha
+  # "beta" maturity APIs should not introduce backwards-incompatible changes
+  # more than once every three months; and when those changes are introduced, the
+  # authors will make a best effort to provide compatibility bridges.
+  - beta
+  # "stable" maturity APIs should not introduce backwards-incompatible changes
+  # more than once every twelve months, and will make every effort to provide
+  # compatibility bridges if at all possible.
+  - stable
+
+
+###############################################################################
+# OPENTELEMETRY COMPONENTS
+#
+# Some OpenTelemetry components run within end-user applications and thus have
+# a footprint across many languages. Naturally, the production and API maturity
+# of these components varies from language to language. Other components run as
+# infrastructure in their own containers or VMs and thus have a single maturity
+# level regardless of the end-user's programming language.
+#
+# The maturity – or even the existence – of the "perLanguageComponents" varies
+# from language to language. These are components that integrate into the
+# actual end-user application runtime in production. See "languages" for the
+# actual maturity levels.
+perLanguageComponents:
+  - traceAPI
+  - metricsAPI
+  - contextAPI
+  - traceSDK
+  - metricsSDK
+  - contextSDK
+  - automaticInstrumentation
+
+# The maturity of the "standaloneComponents" is independent of any particular
+# end-user programming language.
+standaloneComponents:
+  collector:
+    api: beta
+    production: beta
+  operator:
+    api: alpha
+    production: beta
+
+# For each language, we either specify the perLanguageComponents maturity
+# inline, or we specify a single "url" that points to another YAML file with a
+# top-level perLanguageComponents key (presumably in the OpenTelemetry language
+# repository). E.g., something like this:
+#
+# languages:
+#   logo:
+#     url: "https://raw.githubusercontent.com/open-telemetry/opentelemetry-logo/master/maturity.yaml"
+languages:
+  java:
+    traceAPI:
+      api: alpha
+      production: beta
+    metricsAPI:
+      api: alpha
+      production: beta
+    contextAPI:
+      api: alpha
+      production: beta
+    traceSDK:
+      api: alpha
+      production: unstable
+    metricsSDK:
+      api: alpha
+      production: unstable
+    contextSDK:
+      api: alpha
+      production: unstable
+    automaticInstrumentation:
+      api: beta
+      production: unstable
+
+  # TODO once we've converged on the schema for all of this
+  js: ""
+  dotnet: ""
+  ruby: ""
+  erlang: ""
+  python: ""
+  golang:
+    traceAPI:
+      api: beta
+      production: beta
```

Comment:
I'd suggest to include the link to the issues query on GitHub showing what's left for this area. It may be an optional attribute

Reply:
I'm a little hesitant to do this since in the early stages, that list will have lots of errors of omission and could lead to a false sense of security...

Reply:
can we declare it as optional so I can use it for .NET?

Reply:
IMO it'd be good to have a URL field in the schema that points to the repo.